### PR TITLE
Make deal protocol version configurable

### PR DESCRIPTION
**Pull request by en0ma**
This PR's primary goal is to allow operators to choose to send `boost` only, `legacy-markets` only or both (default) deals to miners that support the respective deal protocols. 

I used the opportunity to refactor the code to make deal-making more deterministic. I will create a follow-up PR to extract deal-making and miner selection into their own packages to set the stage for integration and unit test.

I have tested these changes and verified the correct deal protocol version/s are only selected and deals/transfers are made;

- running estuary using `./estuary  --datadir=../dump --logging --deal-protocol-version=v110`
Screenshot 2022-07-26 at 19 37 37

- running estuary using `./estuary  --datadir=../dump --logging --deal-protocol-version=v120`

Screenshot 2022-07-26 at 19 58 29

- running estuary without the flag (default - both deal protocol versions will be enabled) `./estuary  --datadir=../dump --logging`

Screenshot 2022-07-26 at 20 04 13

### CODE REVIEW

**Review by alvin-reyes — APPROVED** [review]
LGTM.

We don't set any `--deal-protocol-version` by default which will force the process to use both protocols.

File: main.go [hunk 1]
```diff
@@ -453,6 +468,10 @@ func main() {
 			Usage: "sets shuttle message handler count",
 			Value: cfg.ShuttleMessageHandlers,
 		},
+		&cli.StringSliceFlag{
+			Name:  "deal-protocol-version",
+			Usage: "sets the deal protocol version. deafults to v110 (go-fil-markets) and v120 (boost)",
```
[the remark this answers is not in the document]
Comment (alvin-reyes) [reply main.go:380]: minor feedback here.

File: replication.go [hunk 2]
```diff
@@ -867,13 +855,12 @@ func (cm *ContentManager) pickMinerDist(n int) (int, int) {
...
 		return 2, n - 2
 	}
-
 	return n - (n / 2), n / 2
 }
 
 const topMinerSel = 15
 
-func (cm *ContentManager) pickMiners(ctx context.Context, cont Content, n int, size abi.PaddedPieceSize, exclude map[address.Address]bool) ([]address.Address, error) {
+func (cm *ContentManager) pickMiners(ctx context.Context, n int, pieceSize abi.PaddedPieceSize, exclude map[address.Address]bool, filterByprice bool) ([]miner, error) {
```
[the remark this answers is not in the document]
Comment (alvin-reyes) [reply replication.go:853]: minor feedback here.

File: config/estuary.go [hunk 3]
```diff
@@ -71,6 +73,10 @@ func NewEstuary(appVersion string) *Estuary {
 			Disable:               false,
 			FailOnTransferFailure: false,
 			Verified:              true,
+			EnabledDealProtocolsVersions: map[protocol.ID]bool{
```
Comment (en0ma) [inline config/estuary.go:76]: @alvin-reyes the default is set here. If the flag is not passed, this will be used

**Review by alvin-reyes — APPROVED** [review]
LGTM.

**Pull request merged**